### PR TITLE
vioser_in_use: add case of vcpu hotplug

### PR DESCRIPTION
**Pull request by xiagao**
Vcpu hotplug test during data transfer via serial.
ID: 1566881
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
@nanliu-r @yfu Call for your review, thanks.

File: qemu/tests/vioser_in_use.py [hunk 1]
```diff
@@ -116,10 +133,20 @@ def run(test, params, env):
...
-    if bg_thread:
-        bg_thread.join(timeout=timeout, suppress_exception=suppress_exception)
-    if vm.is_alive():
-        kill_host_serial_pid(params, vm)
-        if (virtio_serial_file_transfer.transfer_data(params, vm, sender=sender) is
-                not True):
-            test.fail("Serial data transfter test failed.")
+
+    # for vcpu hotplug subtest, only check guest crash.
+    vcpu_hotplug = params.get_boolean("vcpu_hotplug")
```
Comment (nanliu-r) [inline qemu/tests/vioser_in_use.py:138]: If the default bool of `vcpu_hotplug` is false, then there should be `vcpu_hotplug = params.get_boolean("vcpu_hotplug"，False)` .

File: qemu/tests/vioser_in_use.py [hunk 2]
```diff
@@ -45,6 +46,22 @@ def live_migration_guest(test, params, vm, session):
...
 
+@error_context.context_aware
+def vcpu_hotplug_guest(test, params, vm, session):
+    """
+    Vcpu hot plug test.
+    """
+
+    maxcpus = int(params["vcpu_maxcpus"])
+    current_cpus = int(params.get("smp", 2))
+    cpu_hotplug_cmd = params.get("cpu_hotplug_cmd", "")
```
Comment (vivianQizhu) [inline qemu/tests/vioser_in_use.py:57]: I think this is unnecessary in your case, leave `plug_command` as its default value when you call `vm.hotplug_vcpu()` would be okay.

File: qemu/tests/vioser_in_use.py [hunk 3]
```diff
@@ -116,10 +133,20 @@ def run(test, params, env):
...
-                not True):
-            test.fail("Serial data transfter test failed.")
+
+    # for vcpu hotplug subtest, only check guest crash.
+    vcpu_hotplug = params.get_boolean("vcpu_hotplug")
+    if vcpu_hotplug:
+        error_context.context("Check if guest is alive.", logging.info)
+        vm.verify_kernel_crash()
+        session = vm.wait_for_login(timeout=timeout)
+        session.close()
```
Comment (vivianQizhu) [inline qemu/tests/vioser_in_use.py:143]: What do you think if `return` here and remove line 144? Thus narrow down the code hierarchy.

File: qemu/tests/vioser_in_use.py (hunk 2 above)
Comment (xiagao) [reply qemu/tests/vioser_in_use.py:57]: yes, will update.

File: qemu/tests/vioser_in_use.py (hunk 3 above)
Comment (xiagao) [reply qemu/tests/vioser_in_use.py:143]: The method is better than mine, it's ok for me, will update

File: qemu/tests/vioser_in_use.py (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/vioser_in_use.py:138]: Since the default bool value is false it's fine with this way.

**Comment by vivianQizhu** [comment]
@nanliu-r Would you help review and test it? Thanks

**Comment by nanliu-r** [comment]
@xiagao can you offer the latest test results? thanks.

**Comment by xiagao** [comment]
(1/3) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.vioser_in_use.with_vcpu_hotplug.transfer_both: PASS (172.94 s)
 (2/3) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.vioser_in_use.with_vcpu_hotplug.transfer_both: PASS (125.91 s)
 (3/3) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.vioser_in_use.with_vcpu_hotplug.transfer_both: PASS (81.09 s)

**Comment by nanliu-r** [comment]
ACK.

**Pull request merged**